src/bokeh/models/annotations/dimensional.py

```diff
@@ -68,29 +76,113 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
     A subset of units from the basis to avoid.
     """)
 
+    @abstractmethod
+    def is_known(self, unit: str) -> bool:
+        pass
+
 @abstract
-class Metric(Dimensional):
+class CustomDimensional(Dimensional):
+    """ A base class for units of measurement with an explicit basis.
     """
-    A base class defining metric units of measurement.
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+    basis = Required(Dict(String, Tuple(Float, String)), help="""
+    TODO
+    """)
+
+    def is_known(self, unit: str) -> bool:
+        return unit in self.basis
+
+class Metric(Dimensional):
+    """ Model for defining metric units of measurement.
     """
 
     # explicit __init__ to support Init signatures
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        super().__init__(*args, **kwargs)
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+    base_unit = Required(String, help="""
+    TODO
+    """)
 
     ticks = Override(default=[1, 2, 5, 10, 15, 20, 25, 50, 75, 100, 125, 150, 200, 250, 500, 750])
 
+    def is_known(self, unit: str) -> bool:
+        prefixes = ["Q", "R", "Y", "Z", "E", "P", "T", "G", "M", "k", "h", "", "d", "c", "m", "µ", "n", "p", "f", "a", "z", "y", "r", "q"]
+        basis = {f"{prefix}{unit}" for prefix in prefixes}
+        return unit in basis
+
+class ReciprocalMetric(Metric):
+    """ Model for defining metric units of measurement.
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
 class MetricLength(Metric):
+    """ Metric units of length measurement.
     """
-    Units of metric length.
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+    base_unit = Override(default="m")
+
+    exclude = Override(default=["dm", "hm"])
+
+class ReciprocalMetricLength(ReciprocalMetric):
+    """ Metric units of reciprocal length measurement.
     """
 
     # explicit __init__ to support Init signatures
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        super().__init__(*args, **kwargs)
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+    base_unit = Override(default="m")
 
     exclude = Override(default=["dm", "hm"])
 
+class ImperialLength(CustomDimensional):
+    """ Imperial units of length measurement.
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+    basis = Override(default={
+        "in":  ( 1/12, "in" ),
+        "ft":  (    1, "ft" ),
+        "yd":  (    3, "yd" ),
+        "ch":  (   66, "ch" ),
+        "fur": (  660, "fur"),
+        "mi":  ( 5280, "mi" ),
+        "lea": (15840, "lea"),
+    })
```

Comment:
I have no idea what "ch" and "lea" are, comments after each case would be helpful

Reply:
Indeed! Assume ch is the abbrevation for a [chain](https://en.wikipedia.org/wiki/Chain_(unit)) and lea is a [Lea](https://en.wikipedia.org/wiki/Lea_(unit)).